x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/Realm.java

```diff
@@ -144,17 +154,45 @@ public void usageStats(ActionListener<Map<String, Object>> listener) {
         listener.onResponse(stats);
     }
 
+    public RealmRef getRealmRef() {
```

Comment:
Nit: existing getter methods in this class following the Builder type naming convention, e.g. `name()` instead of `getName()`, while this method uses the JavaBean naming convention (`getXxx`).  I am OK with either of them, but would prefer consistency within a single class.